include/seqan3/alphabet/nucleotide/dna15.hpp

```diff
@@ -248,123 +128,70 @@ struct dna15
         'Y'
     };
 
-    //!\copydoc seqan3::dna4::char_to_value
-    static constexpr std::array<internal_type, 256> char_to_value
+    //!\copydoc seqan3::dna4::char_to_rank
+    static constexpr std::array<rank_type, 256> char_to_rank
     {
         [] () constexpr
         {
-            using in_t = internal_type;
-            std::array<in_t, 256> ret{};
+            std::array<rank_type, 256> ret{};
 
             // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
             for (auto & c : ret)
-                c = in_t::UNKNOWN;
+                c = 8; // rank of 'N'
+
+            // reverse mapping for characters and their lowercase
+            for (size_t rnk = 0u; rnk < value_size; ++rnk)
+            {
+                ret[         rank_to_char[rnk] ] = rnk;
+                ret[to_lower(rank_to_char[rnk])] = rnk;
+            }
```

Comment:
I'm surprised that this even works. I had problems in the past to use variables in the same class at declaration time.

Reply:
?